### DISCUSSION TITLE
Need help in PUG SASS Webpack config File

### DISCUSSION THREAD

**Discussion by WINOFFRG** [webpack/webpack#14152] [Q&A]
Hi! I have just started using webpack and want to configure to compile PUG and SASS Files. But facing compilation error, Do pardon me if there is a silly mistake. Here below on left is my project Structure with `webpack.congif.js` file in middle.

Code Input: `src` folder in root directory having `pug` and `sass` folder separately.
Code Output: Expected compiled code in `dist` folder.

![image](https://user-images.githubusercontent.com/61311751/132199954-df997ed3-1d13-43d4-a881-a6cea4373cb3.png)

I have replaced project root address with this **MY_PROJECT_ROOT_DIRECTORY**

Error: 
```

Compiled with problems:X
Error:
Module not found: Error: Can't resolve './src' in 'MY_PROJECT_ROOT_DIRECTORY'

Error:
  Error: Child compilation failed:
  Module not found: Error: Can't resolve 'MY_PROJECT_ROOT_DIRECTORY\src\pug' in 'MY_PROJECT_ROOT_DIRECTORY'
  ModuleNotFoundError: Module not found: Error: Can't resolve 'MY_PROJECT_ROOT_DIRECTORY\src\pug' in 'MY_PROJECT_ROOT_DIRECTORY'
```

Would be very thankful, If someone could help me with this issue.

**Comment by alexander-akait** [comment]
You should use `path.resolve(MY_PROJECT_ROOT_DIRECTORY)`

**Reply by WINOFFRG** [reply]
Hey, Thanks for your response. Till the time I waited for someone's response, I found another way and things now seems to work but yes with few more errors. Here below is the complete code of my `webpack.config.js` file.

```
const webpack = require('webpack');
const path = require('path');
const LiveReloadPlugin = require('webpack-livereload-plugin');
const BrowserSyncPlugin = require('browser-sync-webpack-plugin');

const PATHS = {
    src: path.join(__dirname, 'src'),
    dist: path.join(__dirname, 'dist')
  };
  
const html = PATHS.src + '/views/index.pug';
const HtmlWebpackPlugin = require('html-webpack-plugin');
const { cleanup } = require('browser-sync');

const compileTemplate = (fileName) => {
    return new HtmlWebpackPlugin({
      template: PATHS.src + '/views/' + fileName + '.pug',
      filename: fileName + '.html',
      baseUrl: 'MY_PROJECT_ROOT_DIRECTORY'
    })
  }

module.exports = {
  
    entry: {
        index: PATHS.src + '/index.js',
    },

    output: {
      path: PATHS.dist,
      clean: true
    },
  
    module: {
      rules: [
        {
          test: /\.pug$/,
          use: [
            'pug-loader'
          ]
        },
  
        {
          test: /\.scss|css$/,
          use: [
            'style-loader',
            'css-loader',
            {
              loader: 'sass-loader',
            }
          ]
        },
      ]
    },
  
    resolve: {
      extensions: ['.ts', '.js']
    },

    plugins: [
        new LiveReloadPlugin(),
        new BrowserSyncPlugin ({
          host: 'localhost',
          port: 3000,
        //   proxy: 'http://localhost:8080'
        }),
        compileTemplate('index'),
      ],
    
      mode: 'development',
};
```

The config above opens up the `index.html` or actually `index.pug` file properly in the browser. But I am still having a few doubts, If you could please clear them:
1.  How to link the SCSS Files and get them render too.
2. I am not getting any output in `dist` folder or no where but still the files get displayed in browser.
3. After saving a file it takes 7-10sec for the `BrowserSync` to reload the file.
4. What is the use of `index.js` in my `src` folder? It's empty but if I remove it the compiler throws error.

**Reply by alexander-akait** [reply]
Please use stackoverflow for these questions
